Build failure on GHC 8.10 (win64)

I'm following along some blog posts and saw that 8.10 is when we get cost-free polysemy abstractions, so I dutifully grabbed it, and cabal, and manually fixed up my path, got `polysemy` installed and imported into a `ghci` session, but when installing the plugin I got the following errors:

    > cabal install --lib polysemy-plugin

    Preprocessing library for polysemy-plugin-0.2.5.0..
    Building library for polysemy-plugin-0.2.5.0..
    [1 of 7] Compiling Paths_polysemy_plugin ( dist\build\autogen\Paths_polysemy_plugin.hs, dist\build\Paths_polysemy_plugin.o )
    [2 of 7] Compiling Polysemy.Plugin.Fundep.Stuff ( src\Polysemy\Plugin\Fundep\Stuff.hs, dist\build\Polysemy\Plugin\Fundep\Stuff.o )
    [3 of 7] Compiling Polysemy.Plugin.Fundep.Unification ( src\Polysemy\Plugin\Fundep\Unification.hs, dist\build\Polysemy\Plugin\Fundep\Unification.o )
    
    src\Polysemy\Plugin\Fundep\Unification.hs:133:23: error:
        Not in scope: type constructor or class `Ct'
        |
    133 |     -> [(Unification, Ct)]
        |                       ^^
    
    src\Polysemy\Plugin\Fundep\Unification.hs:134:25: error:
        Not in scope: type constructor or class `Ct'
        |
    134 |     -> ([Unification], [Ct])
        |                         ^^
    cabal: Failed to build polysemy-plugin-0.2.5.0. See the build log above for
    details.

_Edit: I tried checking this repo out @ 016c16fbb1b57a0d728e57e2cf8e36453e8edd8d thinking the version in Hackage might be different, but I got the same result._

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,7 @@ cabal: "3.0"
 
 matrix:
   include:
+    - ghc: "8.10.1"
     - ghc: "8.8.1"
     - ghc: "8.6.5"
     - ghc: "8.4.4"

--- a/polysemy-plugin/src/Polysemy/Plugin.hs
+++ b/polysemy-plugin/src/Polysemy/Plugin.hs
@@ -64,6 +64,7 @@ module Polysemy.Plugin
 import Polysemy.Plugin.Fundep
 #if __GLASGOW_HASKELL__ >= 810
 import Polysemy.Plugin.Phases
+import Data.Bool (bool)
 #endif
 
 import GhcPlugins

--- a/polysemy-plugin/src/Polysemy/Plugin/Fundep.hs
+++ b/polysemy-plugin/src/Polysemy/Plugin/Fundep.hs
@@ -46,6 +46,7 @@ import           Polysemy.Plugin.Fundep.Utils
 import           TcEvidence
 import           TcPluginM (TcPluginM, tcPluginIO)
 import           TcRnTypes
+import           Constraint
 import           TcSMonad hiding (tcLookupClass)
 import           Type
 

--- a/polysemy-plugin/src/Polysemy/Plugin/Fundep.hs
+++ b/polysemy-plugin/src/Polysemy/Plugin/Fundep.hs
@@ -1,5 +1,6 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE ViewPatterns               #-}
+{-# LANGUAGE CPP                        #-}
 
 ------------------------------------------------------------------------------
 -- The MIT License (MIT)
@@ -46,7 +47,9 @@ import           Polysemy.Plugin.Fundep.Utils
 import           TcEvidence
 import           TcPluginM (TcPluginM, tcPluginIO)
 import           TcRnTypes
+#if __GLASGOW_HASKELL__ >= 810
 import           Constraint
+#endif
 import           TcSMonad hiding (tcLookupClass)
 import           Type
 

--- a/polysemy-plugin/src/Polysemy/Plugin/Fundep/Unification.hs
+++ b/polysemy-plugin/src/Polysemy/Plugin/Fundep/Unification.hs
@@ -3,7 +3,8 @@ module Polysemy.Plugin.Fundep.Unification where
 import           Data.Bool
 import           Data.Function (on)
 import qualified Data.Set as S
-import           TcRnTypes
+import           Constraint
+
 import           Type
 
 

--- a/polysemy-plugin/src/Polysemy/Plugin/Fundep/Unification.hs
+++ b/polysemy-plugin/src/Polysemy/Plugin/Fundep/Unification.hs
@@ -1,9 +1,15 @@
+{-# LANGUAGE CPP                       #-}
+
 module Polysemy.Plugin.Fundep.Unification where
 
 import           Data.Bool
 import           Data.Function (on)
 import qualified Data.Set as S
+#if __GLASGOW_HASKELL__ >= 810
 import           Constraint
+#else
+import           TcRnTypes
+#endif
 
 import           Type
 

